src/bogart/AS_BAT_SplitTig.H

```diff
@@ -0,0 +1,153 @@
+
+/******************************************************************************
+ *
+ *  This file is part of canu, a software program that assembles whole-genome
+ *  sequencing reads into contigs.
+ *
+ *  This software is based on:
+ *    'Celera Assembler' r4587 (http://wgs-assembler.sourceforge.net)
+ *    the 'kmer package' r1994 (http://kmer.sourceforge.net)
+ *
+ *  Except as indicated otherwise, this is a 'United States Government Work',
+ *  and is released in the public domain.
+ *
+ *  File 'README.licenses' in the root directory of this distribution
+ *  contains full conditions and disclaimers.
+ */
+
+#ifndef INCLUDE_AS_BAT_SPLIT_TIG
+#define INCLUDE_AS_BAT_SPLIT_TIG
+
+#include "AS_BAT_ReadInfo.H"
+#include "AS_BAT_OverlapCache.H"
+#include "AS_BAT_BestOverlapGraph.H"
+#include "AS_BAT_AssemblyGraph.H"
+#include "AS_BAT_Logging.H"
+
+#include "AS_BAT_Unitig.H"
+
+#include "AS_BAT_MarkRepeatReads.H"
+
+#include "intervalList.H"
+#include "stddev.H"
+
+#include <vector>
+
+using namespace std;
+
+
+//  We can split a tig based on a list of repeat/unique intervals, or based
+//  on a list of read ends.
+//
+//  A list of breakPointCoords will split a tig into multiple tigs (and label
+//  those new tigs as repeat or unique) based on how reads intersect the supplied
+//  intervals.  A read must be fully contained in a repeat interval for it to be
+//  moved to the new tig, while it only needs to touch a unique interval for it
+//  to end up there.
+//
+//  A list of breakPointEnd will slice the tig at that read end.
+//    _bgn == true  -> reads that begin at after position are in the region
+//    _end == false -> reads that end before position are in the region
+
+class breakPointCoords {
+public:
+  breakPointCoords(int32 bgn, int32 end, bool rpt=false) {
+    _bgn = bgn;
+    _end = end;
+    _rpt = rpt;
+  };
+  ~breakPointCoords() {
+  };
+
+  bool    operator<(breakPointCoords const &that) const {
+    return(_bgn < that._bgn);
```

Comment:
Is it ok not to use other fields if begins match? Essentially they will be considered equal, right?